File: DesignPatternsExercises/java/MediatorDesignPattern/Exercise/CurrencyConverter.java
import java.util.HashMap;
import java.util.Map;

public class CurrencyConverter {
	private static Map<String, Float> conversionToDollarRates = new HashMap<>();
	__________(3)__________ {
		conversionToDollarRates.put("USD", (float) 1);
		conversionToDollarRates.put("EUR", (float) 0.9378);
		conversionToDollarRates.put("GBP", (float) 0.7974);
		conversionToDollarRates.put("INR", (float) 64.54);
	}

	public static float convert(String currency, float amount) {
		return amount / conversionToDollarRates.get(currency);
	}
}

File: DesignPatternsExercises/ruby/mediator_design_pattern/solution/currency_converter.rb
class CurrencyConverter
  # __________(3)__________ = { USD: 1, EUR: 0.9378, GBP: 0.7974, INR: 64.54 }
  @@conversion_to_dollar_rates = { USD: 1, EUR: 0.9378, GBP: 0.7974, INR: 64.54 }

  # def __________(4)__________.convert(currency, amount)
  def self.convert(currency, amount)
    amount / @@conversion_to_dollar_rates[currency]
  end
end
